docs/logging.md

```diff
@@ -0,0 +1,75 @@
+# Logging
+
+Thanos supports query logging via flags.
+
+Components are configured using `--request.logging-config-file` to reference to the configuration file or `--request.logging-config` as YAML directly.
+
+## Configuration
+
+Configuration can be supplied for either gRPC, HTTP or both. Options can be supplied globally, which is applied to both gRPC and HTTP, or individually to either or both protocols.
+
+### HTTP
+
+```yaml
+http:
+  config:
+    - path: /api/v1/query
+      port: 10904
```

Comment:
What's this port config for?

Reply:
That would be coming from https://github.com/thanos-io/thanos/blob/main/pkg/logging/yaml_parser.go#L34

However I did not look too deeply into the implementation. Just double checked the original example https://gist.github.com/yashrsharma44/02f5765c5710dd09ce5d14e854f22825 against what was exposed by those types.

Reply:
It would be nice to document this field in more detail. Maybe @yashrsharma44 can help answer this. I feel confused when I saw this config TBH.

Reply:
Agree, I just don't have enough context to provide more info. I had a look at https://github.com/thanos-io/thanos/pull/3862 but it didn't tell me much.

Reply:
I believe this is to specify which port to log the requests on - this might make sense in the context of components exposing multiple HTTP endpoints (e.g. receiver). But I'm also thinking if we could make the user experience here more intuitive and somehow figure out which HTTP server it is based on paths? :thinking:

Reply:
I am not sure why it has to be so complicated personally, but maybe missing some context.

At https://github.com/thanos-io/thanos/blob/main/pkg/logging/http.go#L63 we purposely check the port and append it to the decider url. So a request for example to `http:localhost:9090/api/v1/query` comes in initially as `/api/v1/query` before we bolt on the port and pass it to the "decider" function. Could it not just be excluded entirely?

Reply:
Yeah, I agree. Exact match is inflexible.

Reply:
we can avoid the exact match and allow essentially a wildcard match by not configuring anything. Same applies for gRPC. I've documented as such

Reply:
I'm not able to get this config section to work. Without config, I can see three HTTP log entries:

GET /api/v1/query_exemplars?query=<query>
GET /api/v1/query?query=<query>
GET /api/v1/query_range?query=<query>

but these are removed when I add:

```
config:
    - path: /api/v1/query
      port: 10902
    - path: /api/v1/query_exemplars
      port: 10902
    - path: /api/v1/query_range
      port: 10902
```

10902 being the http port. Have any of you got this working, if so, what did you do differently? Maybe we can use this to improve the docs and make this a bit more clear?

Reply:
@dwilliams782 it is looking for an exact match. regex matching etc is not supported - I've documented the behaviour on https://github.com/thanos-io/thanos/pull/4934/files#diff-1a5e57d66b08bec5a71860d67f06912a004f03368c36910099d4ffae12970c7dR61

Omitting the config will match all as you have seen.

Reply:
@PhilipGough how do we know what an exact match is? AFAIK my config above should be an "exact" match, in that, they are the three endpoints being emitted (when running without config) and that is my HTTP port. How else can I diagnose this?

Reply:
@dwilliams782 - the exact match would include any query params etc - I think it would be beneficial to be able to add `\/api\/v1\/query\?.+` for example, but that is for another issues.

And I am sorry I don't have all the answers since I put this together by stumbling through the same pain points as you are right now :)

My diagnosis was done by running locally and outputting logs and stepping through the debugger - not aware of a better way.

Reply:
Ah I see! Perhaps your example of using config should specify that any parameters need to match too? That wasn't obvious to me.

Yeah, I really appreciate your efforts here. I'm not expecting you to know the answers, but hopefully asking the questions can help improve the documentation when we do stumble on the answers. Cheers!

Reply:
Yeah no worries, good point, I can add a couple of examples with query params to the example conf and it might make it more obvious

By the way @matej-g has created https://github.com/thanos-io/thanos/issues/4961 to track some of the issues and figure out improvements to this so might be worth keeping an eye on that going forward